src/overlays/actors/ovl_Bg_Spot15_Rrbox/z_bg_spot15_rrbox.c

```diff
@@ -21,45 +39,343 @@ const ActorInit Bg_Spot15_Rrbox_InitVars = {
     (ActorFunc)BgSpot15Rrbox_Update,
     (ActorFunc)BgSpot15Rrbox_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot15_Rrbox/func_808B3960.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot15_Rrbox/func_808B39E8.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 1000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 500, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),
+};
+
+Vec3f D_808B45C4[] = {
+    { 770.0f, 1490.0f, -299.0f },
+    { 770.0f, 1550.0f, -299.0f },
+};
+
+Vec3f D_808B45DC[] = { { 29.99f, 0.01f, -29.99f },
+                       { -29.99f, 0.01f, -29.99f },
+                       { -29.99f, 0.01f, 29.99f },
+                       { 29.99f, 0.01f, 29.99f },
+                       { 0.0f, 0.01f, 0.0f } };
+
+s32 D_808B4618[] = { 0, 0 };
+
+void func_808B3960(BgSpot15Rrbox* this, GlobalContext* globalCtx, s32* arg2, s32 flags) {
+    s32 temp1 = 0xAB;
+    s32 sp30 = 0;
+    u32 temp2 = 0x32;
+
+    DynaPolyInfo_SetActorMove(&this->dyna, flags);
+    DynaPolyInfo_Alloc(arg2, &sp30);
+
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, sp30);
+
+    if (this->dyna.dynaPolyId == temp2) {
+        osSyncPrintf("Warning : move BG 登録失敗(%s %d)(name %d)(arg_data 0x%04x)\n", "../z_bg_spot15_rrbox.c", temp1,
```

Comment:
a temp for the line number (which also should be decimal) looks very unlikely, can you just put the line number here and make temp1 just padding?

Reply:
Made both these variables pads instead.